classes/abstracts/ActionScheduler.php

```diff
@@ -181,21 +185,18 @@ public static function init( $plugin_file ) {
 	}
 
 	/**
-	 * Issue deprecated warning if an Action Scheduler function is called in the shutdown hook.
+	 * Check whether the AS data store has been initialized.
 	 *
 	 * @param string $function_name The name of the function being called.
+	 * @return bool
 	 */
-	public static function check_shutdown_hook( $function_name ) {
-		if ( 'shutdown' === current_filter() ) {
-			$message = sprintf(
-				/* translators: $1: open code tag, $2: function name, $3: close code tag. */
-				__( 'Action Scheduler function %1$s%2$s%3$s should not be used within the WordPress %1$sshutdown%3$s hook.', 'action-scheduler' ),
-				'<code>',
-				esc_attr( $function_name ) . '()',
-				'</code>'
-			);
-			_deprecated_hook( 'shutdown', 'Action Scheduler 3.0', 'init', $message );
+	public static function is_initialized( $function_name ) {
+		if ( ! self::$data_store_initialized ) {
```

Comment:
It would be good to make `$function_name` optional and only log the deprecated notice if `$function_name` has some value. This is because `ActionScheduler::is_initialized()` is a useful utility for 3rd party code to check if AS is safe to use on `'shutdown'` or other hooks (instead of `did_action( 'init' )`, which requires the 3rd party knowing more about how AS initializes things).

Reply:
Good suggestion. Done in https://github.com/woocommerce/action-scheduler/pull/546/commits/6bbe5b6addfc3546a8b2f7e8725756eb65ea4a4e